src/runtime/stream.rs

```diff
@@ -277,11 +275,21 @@ impl AsyncWrite for AsyncTcpStream {
 
 impl TokioAsyncRead for AsyncTcpStream {
     fn poll_read(
-        self: Pin<&mut Self>,
+        mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-        buf: &mut [u8],
-    ) -> Poll<tokio::io::Result<usize>> {
-        AsyncRead::poll_read(self, cx, buf)
+        buf: &mut ReadBuf,
+    ) -> Poll<tokio::io::Result<()>> {
+        match self.deref_mut() {
+            #[cfg(feature = "tokio-runtime")]
+            Self::Tokio(ref mut inner) => Pin::new(inner).poll_read(cx, buf),
+
+            #[cfg(feature = "async-std-runtime")]
+            Self::AsyncStd(ref mut inner) => {
+                use tokio_util::compat::FuturesAsyncReadCompatExt;
```

Comment:
Unfortunately `tokio_util` does not provide a one-off adapter function for the reverse direction, so to implement tokio's `AsyncRead` on an `async_std` stream, we need to wrap it in the `Compat` type that tokio_util provides.